include/arch/arm/aarch32/exc.h

```diff
@@ -73,25 +73,45 @@ GTEXT(z_arm_exc_exit);
 extern "C" {
 #endif
 
+struct __basic_sf {
+	sys_define_gpr_with_alias(a1, r0);
+	sys_define_gpr_with_alias(a2, r1);
+	sys_define_gpr_with_alias(a3, r2);
+	sys_define_gpr_with_alias(a4, r3);
+	sys_define_gpr_with_alias(ip, r12);
+	sys_define_gpr_with_alias(lr, r14);
+	sys_define_gpr_with_alias(pc, r15);
+	uint32_t xpsr;
+};
+
 struct __esf {
```

Comment:
Do we really need to keep __esf and __extended_esf?

could we try to use only the "extended" esf where we clearly document that the extra information is at the end?

Otherwise I am OK with this re-factoring.